tsdb/engine/tsm1/file_store.go

```diff
@@ -18,9 +18,14 @@ import (
 	"github.com/influxdata/influxdb/tsdb"
 )
 
+// currentTempDirID is used to create unique directories for snapshotting
+// This is ephemeral, and resets every time the service restarts
+// All old tmp directories are deleted on startup, so we ensure no collisions
+var currentTempDirID int
```

Comment:
Can you move this onto `FileStore`?  `FileStore` manages the shard TSM file paths and it's a little odd to have a package variable protected by an instance of `FileStore`.  There is only ever 1 `FileStore` for a shard at any  given time.

Also, we shouldn't rely on the dir always getting deleted (which could fail).  So it should scan the dir when opening to find the max ID similar to how we load shards.